### PR TITLE
fix(ts-xsd): quote XML property names in generated types

### PR DESCRIPTION
**Pull request by Octember**
## Summary
Valid XML names containing dots or hyphens currently produce invalid TypeScript, for example `MZ.Sequence.EditingModeGUID: string`.

Quote names that are not valid identifiers when emitting interface properties, flattened properties, and root-element wrappers. Ordinary identifiers remain unchanged. No API or dependency changes.

## Verification
- Added regressions that generate both interfaces and flattened types from XSD, compile a typed consumer, and check that exact property names survive.
- All 472 package tests pass; the new regressions fail before the fix.
- Build, source typecheck, lint, and formatting pass.
- The separate test-typecheck target has a pre-existing error in `tests/integration/codegen-infer.test.ts:151`: `personSchema` is referenced but declared as `_personSchema`. Left unchanged.

---
## Summary by cubic
Fixes `ts-xsd` codegen so XSD property names containing dots or hyphens produce valid TypeScript. Previously names like `MZ.Sequence.EditingModeGUID` were emitted unquoted as invalid identifiers; now names that aren't valid identifiers are quoted in interfaces, flattened types, and root-element wrappers, while ordinary identifiers remain unchanged. No API or dependency changes.

## Verification
- Added regressions that generate both interfaces and flattened types from XSD, compile a typed consumer, and check that exact property names survive.
- All 472 package tests pass; the new regressions fail before the fix.
- Build, source typecheck, lint, and formatting pass.
- The separate test-typecheck target has a pre-existing error in `tests/integration/codegen-infer.test.ts:151`: `personSchema` is referenced but declared as `_personSchema`. Left unchanged.

<sup>Written for commit cb18b3591e29d97ed4ba49a3641a68776c107ed9. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/abapify/adt-cli/pull/192?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Bug Fixes**
  * Generated TypeScript interfaces now preserve XML element and attribute names containing dots or hyphens while keeping the output valid TypeScript.
  * Improved generated property names for both flattened and nested schema representations.

___

## **CodeAnt-AI Description**
Generate valid TypeScript for XML names containing dots and hyphens

### What Changed
- XML element, attribute, and nested property names that are not valid TypeScript identifiers are now quoted in generated types.
- Root wrappers and flattened types preserve names such as `adobe.project`, `MZ.Sequence.EditingModeGUID`, and `preview-codec` for bracket notation access.
- Added coverage confirming generated types compile and retain the exact XML property names.

### Impact
`✅ Valid generated TypeScript for dotted and hyphenated XML names`
`✅ Preserved XML property names`
`✅ Reliable bracket-notation access to generated fields`
<details><summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

### CODE REVIEW

**Review by ThePlenkov — COMMENTED** [review]
## Review

CI is fully green (SonarCloud, Codacy, CodeRabbit, CodeScene, Analyze — all pass). The PR is small (84 additions, 3 deletions across 2 files) and fixes a real bug: XSD property names containing dots or hyphens produced invalid TypeScript.

### Files changed summary
- `packages/ts-xsd/src/codegen/ts-morph.ts`: +9/-3 — new `propertyName()` helper + 3 application sites
- `packages/ts-xsd/tests/unit/interface-generator.test.ts`: +75/-0 — regression tests

### Findings

#### Blocking
None.

#### Important — not blocking merge
None.

#### Low — nits / follow-ups

1. **CodeRabbit's Other_ID_Start/Continue comment is technically incorrect** — The regex `\p{ID_Start}` and `\p{ID_Continue}` already includes `Other_ID_Start` (℘, U+2118) and `Other_ID_Continue` (·, U+00B7). I verified at runtime: `a·b` matches the regex and is a valid JS identifier. The suggestion to "update the predicate" is unnecessary. That said, adding explicit tests for Unicode identifiers would be a worthwhile follow-up.

2. **Duplication with existing `isValidIdentifier`** — `generate.ts:468` and `raw-schema.ts:567` already have their own `isValidIdentifier` (ASCII-only regex + reserved-word check). This PR introduces a third, more correct Unicode-aware version. A follow-up could consolidate into a shared utility. Not blocking — the functions live in different modules with different semantics.

3. **`toInterfaceName` does not sanitize invalid identifiers** (pre-existing, out of scope) — If a complexType or inline element name contains dots/hyphens, `toInterfaceName` (line 505) produces an invalid TS identifier (e.g. `Adobe.projectType`). The PR's test only covers named complexType references (`type="Example"`), not inline complexTypes with dotted names. Worth a follow-up issue.

4. **`[false, true] as const` on the test tuple** — Style nit from CodeRabbit, not blocking.

### Positive observations
- The regex `/^[\p{ID_Start}$_][\p{ID_Continue}$\u200C\u200D]*$/u` is the precise ECMAScript identifier specification, Unicode-aware.
- `JSON.stringify` is the correct quoting mechanism: properly escapes quotes, backslashes, and control characters.
- Tests are thorough: both flatten modes, dotted and hyphenated names for elements and attributes, compilation of a typed consumer, and verification that exact property names survive.
- The pre-existing `test-typecheck` error (`_personSchema` vs `personSchema` at line 151) is correctly identified and left unchanged — confirmed pre-existing on `main`.
- All three emission paths are covered: flattened types (line 367), interface properties (line 698), root type wrappers (line 977). Indirect paths (addElementProperty, collectAttributes) feed into line 698 — also covered.

### Open questions
None blocking.

### Recommendation
**Approve.** The fix is correct, well-tested, and appropriately scoped. All findings are Low-severity follow-ups.

File: packages/ts-xsd/src/codegen/ts-morph.ts [hunk 1]
```diff
@@ -65,6 +65,12 @@ export interface SchemaSourceFileResult {
 // Helpers
 // =============================================================================
 
+function propertyName(name: string): string {
+  return /^[\p{ID_Start}$_][\p{ID_Continue}$\u200C\u200D]*$/u.test(name)
```
[the remark this answers is not in the document]
Comment (ThePlenkov) [reply packages/ts-xsd/src/codegen/ts-morph.ts:69]: Verified at runtime — this finding is a false positive.

File: packages/ts-xsd/tests/unit/interface-generator.test.ts [hunk 2]
```diff
@@ -2,6 +2,81 @@ import { describe, it } from 'node:test';
 import assert from 'node:assert';
 import { generateInterfaces } from '../../src/codegen/interface-generator';
 import type { Schema } from '../../src/xsd/types';
+import { parseXsd } from '../../src/xsd/parse';
+
+describe('XML property names', () => {
+  for (const flatten of [false, true]) {
```
[the remark this answers is not in the document]
Comment (ThePlenkov) [reply packages/ts-xsd/tests/unit/interface-generator.test.ts]: Fixed in commit cb18b359 — added `as const` to `[false, true]` to align with the package's `as const` convention.
